### ISSUE TITLE
using assembly Path do not support environment variables

### ISSUE THREAD

**Issue by ichat001** [PowerShell/PowerShell#21457]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

using assembly Path do not support environment variables, I'm a beginner in PowerShell, and I'm not sure if this is an error.

### Expected behavior

```console
using assembly $HOME\Desktop\WebDriver.dll
```

### Actual behavior

```console
ERROR: At Line: 1
ERROR: |
ERROR: |  using assembly $HOME\Desktop\WebDriver.dll
ERROR: |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
ERROR: | '$HOME\Desktop\WebDriver.dll' is not a valid value for using name.
```

### Environment data

```powershell
$HOME
```

**Comment by ThomasNieto** [comment]
I'm not too familiar with this part but I do know the `using` keywords are a parse time construct rather than runtime so I expect variables wouldn't be expanded.

**Comment by rhubarb-geek-nz** [comment]
> using assembly $HOME\Desktop\WebDriver.dll

$HOME is a 'PowerShell variable' not an 'environment variable'. In PowerShell the HOME environment variable would be $env:HOME

**Comment by ichat001** [comment]
> > using assembly $HOME\Desktop\WebDriver.dll
> 
> $HOME is a 'PowerShell variable' not an 'environment variable'. In PowerShell the HOME environment variable would be $env:HOME

There is no $env:HOME variable; it should be $env:USERPROFILE

**Comment by jborean93** [comment]
If you need runtime support for loading an assembly then you can use `Add-Type`. Unfortunately as mentioned above `using` is a parse time component so it supporting vars would be something that couldn't be fully supported outside of maybe a whitelist set of variable.

**Comment by ichat001** [comment]
> If you need runtime support for loading an assembly then you can use `Add-Type`. Unfortunately as mentioned above `using` is a parse time component so it supporting vars would be something that couldn't be fully supported outside of maybe a whitelist set of variable.

Because the 'using' command needs to be placed at the beginning of the code, does that also mean that no variables can be used?

**Comment by jborean93** [comment]
`using ...` is processed when the code is parsed which is before any of it is run. The runtime equivalent action is `Add-Type`.

**Comment by mklement0** [comment]
To provide additional context:

* `using assembly` currently provides _no_ advantage over `Add-Type`: on the contrary, it is less flexible, due to the inability to use variables (of any kind)

* `using assembly` _should_ provide an advantage over `Add-Type`, however, namely making the target assembly's types available at _parse time_, so that these types can be used in PowerShell `class` definitions, but this has never been implemented - see #18482 

* `using assembly` _should_ accept well-known assembly names such as `System.Windows.Forms`, but that is broken in PS Core (it works in Windows PowerShell) - see #11856 

In other words: you're currently better off using `Add-Type` anyway.

If/when the parse-time loading of types for `using assembly` gets implemented and this functionality is required in a given situation, the only dynamic aspect supported in an assembly path is to make it a _relative_ path,  in which case the assembly is looked for relative to the _caller_'s location (rather than the current location).
